### DISCUSSION TITLE
Welcome to helios Discussions!

### DISCUSSION THREAD

**Discussion by mengqi-z** [JGCRI/helios#8] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ansir106** [comment]
Dear JGCRI Team,

My name is Ansir Ilyas, and I am a Postdoctoral Fellow at King Abdullah University of Science and Technology (KAUST) in Saudi Arabia, working in the Systems Science Lab lead by Professor Yoshihide Wada.

I am currently working on calculating HDD/CDD using ISIMIP temperature and population data (https://data.isimip.org/search/tree/ISIMIP3b/InputData/socioeconomic/pop/2015soc/). However, I am encountering the following error when trying to process the ISIMIP population data with the Helios package. This error occurred in read_population function.

Could you please assist in identifying the issue? I downloaded the latest version of the Helios package from GitHub last week.

Thank you very much for your support.

Best,
Ansir Ilyas

Error in `dplyr::mutate()`:
ℹ In argument: `value = as.numeric(tidyr::replace_na(value, 0))`.
Caused by error:
! `value` must be size 259200 or 1, not 0.

**Reply by mengqi-z** [reply]
Hi Ansir,

`Helios` does not currently support the ISIMIP population data format. However, it is compatible with the [SEDAC](https://sedac.ciesin.columbia.edu/data/set/popdynamics-1-8th-pop-base-year-projection-ssp-2000-2100-rev01) population data format. To use your own population data with `Helios`, convert your NetCDF file into a CSV format. The CSV should be structured with columns as follows: `[latitude, longitude, year1, year2, year3, ...]`. Once converted, you can provide the path to your population CSV file to `Helios`.

**Reply by ansir106** [reply]
Dear Mengqi,
Thank you very much for your reply. I am currently working with the GCAM model, which operates on five-year time steps. The Helios package is working perfectly with SEDAC data on my computer, but that data is only available in ten-year intervals. For this reason, I was considering using ISIMIP data. However, I appreciate your comment and will follow your suggestion.

**Comment by ansir106** [comment]
Dear JGCRI Team,

I’m using ISIMIP daily temperature data to calculate HDD/CDD for the period from 2015 to 2100. The data is provided in 9 separate .nc files. Is it possible to load all these files at once and run Helios to calculate HDD/CDD for the entire period in one go? This approach would also generate the necessary XML files. Otherwise, I would have to run the calculations separately for each 5-year interval and manually update the XML files to cover 2015 to 2100, which would be very time-consuming.

If loading all the files at once isn’t possible, could you please suggest an alternative way to produce the XML files for the full time period?

Your help would be greatly appreciated.

Best,
Ansir Ilyas
Postdoc fellow
KAUST, KSA

**Reply by mengqi-z** [reply]
Hi Ansir,

Yes, `helios` can process multiple temperature datasets at once. You can provide the temperature files as a list, such as `c('file_2015_2020.nc', 'file_2021_2030.nc', ..., 'file_2091_2100.nc')`, and pass them to the `ncdf` argument. Please note that due to the large size of the climate data, running multiple climate files could take a while.

Thanks,
Mengqi

**Reply by ansir106** [reply]
Dear Mengqi,

Thank you very much; this is really helpful. I have passed all the temperature files to the hdcd argument, and it is now working perfectly. Thank you again.

Best,
Ansir

**Comment by ansir106** [comment]
Dear JGCRI team, Dear Mengqi,

I am working on the calculation of HDD/CDD for future periods using ISIMIP data. Helios provided all 32 regions when I calculated HDD/CDD for the historical data, but for the future data, it only gave me a few regions (not all 32). I haven't been able to figure out the issue. Could you please help me with this?

I am using SSP data for the period 2015-2100.

Best,
Ansir

**Reply by mengqi-z** [reply]
Hi Ansir,

If the data spans the entire globe, Helios should be able to calculate for all 32 regions. I recommend checking both the temperature and population data in the regions where Helios is not currently producing results to identify any unexpected values (e.g., NA values).

Thanks,
Mengqi
